Feature/30mm ammo

AdozenBABYdogs opened the pull request:
**When merged this pull request will:**
This merge will add two ammo types to 30mm
+ AP Added
+ HEDP Added
* Pylon Presets tweaked

Commit:
Added 30mm HEI and 30mm AP ammo types

Commit:
Added 30mm HEI and 30mm AP ammo types

Commit:
Added 30mm HEI and 30mm AP ammo types

Commit:
Added 30mm HEI and 30mm AP ammo types

Commit:
Adjusted 30mm HEDP and AP ammo

AP made slightly stronger, HE made slightly worse

Commit:
Adjusted 30mm HEDP and AP ammo

AP made slightly stronger, HE made slightly worse

Commit:
Adjusted 30mm HEDP and AP ammo

AP made slightly stronger, HE made slightly worse

Commit:
Adjusted 30mm HEDP and AP ammo

AP made slightly stronger, HE made slightly worse

Commit:
Adjustments to fix error "HEI not found" on the DAP

Ready for QA testing

Commit:
Adjustments to fix error "HEI not found" on the DAP

Ready for QA testing

Commit:
Adjustments to fix error "HEI not found" on the DAP

Ready for QA testing

Commit:
Adjustments to fix error "HEI not found" on the DAP

Ready for QA testing

Commit:
Adjustments to post merge spaghetti 

Fixed "vtx_300Rnd_30mm_shells_hedp" being defined twice

Commit:
Adjustments to post merge spaghetti 

Fixed "vtx_300Rnd_30mm_shells_hedp" being defined twice

Commit:
Adjusted Presets

Preset adjustments made, fixed merge spaghetti

Commit:
Adjusted Presets

Preset adjustments made, fixed merge spaghetti

AdozenBABYdogs commented on addons/MH60S/config/Pylons.hpp:
```diff
@@ -6,7 +6,7 @@ class TransportPylonsComponent {
             attachment = "VTX_4Rnd_ACE_Hellfire_AGM114K";
             priority = 5;
             hardpoints[] = {
-              "DAR", "DAGR", "B_SHIEKER", "20MM_TWIN_CANNON",
```
Forgotten Comma

ampersand38 replied:
this file doesn't looks like it should have any changes from main.

AdozenBABYdogs commented on addons/MH60S/config/Pylons.hpp:
```diff
@@ -15,7 +15,7 @@ class TransportPylonsComponent {
             attachment = "VTX_PylonRack_M261_APKWS";
             mirroredMissilePos = 1;
             hardpoints[] = {
-              "DAR", "DAGR", "B_SHIEKER", "20MM_TWIN_CANNON",
```
Forgoten Comma x2

ampersand38 commented on addons/uh60_weapons/config/cfgAmmo.hpp:
```diff
@@ -59,4 +61,34 @@ class CfgAmmo {
   	};
   };
 
+  class vtx_300Rnd_30mm_shells_ap : B_30mm_MP_Tracer_Green {
```
ammo class shouldn't have round count

ampersand38 commented on addons/uh60_weapons/config/cfgMagazines.hpp:
```diff
@@ -28,14 +28,14 @@ class cfgMagazines {
     pylonWeapon = "vtx_hellfire_launcher_N";
   };
-  class VTX_M230_Chaingun_L: VehicleMagazine {
+  class VTX_M230_Chaingun_L_HEDP: VehicleMagazine {
```
keep the old weapon and magazine class names please, and have them use the new ammo. adding ap version sounds great.

ampersand38 commented on addons/uh60_weapons/config/cfgWeapons.hpp:
```diff
@@ -15,13 +15,12 @@ class cfgWeapons {
+      magazines[] = {"VTX_M230_Chaingun_L_HEDP", "VTX_M230_Chaingun_R_HEDP"};
       class player: player {
-        displayName = "M230";
+        displayName = "M230 HEDP";
```
please fix the merge issue with the `reloadTime` from main

Commit:
Merge remote-tracking branch 'origin/feature/30mm_ammo' into feature/30mm_ammo

Commit:
Update to Classnames, Reload Time

Reverted cfgMagazines classnames to prevent breaking prior scripts.  Adjusted ammo class names to be in line with Arma 3 naming convention.  Fixed Reload Time back to prior number.  Fixed Comma Error

ampersand38 commented on addons/UH60/config/cfgWeapons.hpp:
```diff
@@ -1,5 +1,6 @@
 class cfgWeapons {
     class LMG_Minigun;
+    class gatling_20mm;
```
```suggestion

```
Doesn't seem to be needed?

ampersand38 commented on addons/MH60S/config/Pylons.hpp:
```diff
@@ -41,6 +41,8 @@ class TransportPylonsComponent {
         DUMMY_PYLON(dummy14)  //Right HMD Color B
         DUMMY_PYLON(dummy15)  //Right HMD Brightness
     };
+    //TODO
+    //presets commented out for work with Ampersand
```
```suggestion
```
Seahawk should have the 20mm pod since the M230 clips through the cabin

ampersand38 commented on addons/uh60_weapons/config/cfgWeapons.hpp:
```diff
@@ -52,12 +50,20 @@ class cfgWeapons {
+        displayName = "M230 AP";
+      };
+    };
+	class vtx_apkws: ace_missileguidance_dagr {
```
```suggestion
	class vtx_apkws: vtx_dagr {
```

ampersand38 commented on addons/uh60_weapons/config/cfgMagazines.hpp:
```diff
@@ -70,6 +83,34 @@ class cfgMagazines {
     model = QPATHTOF(fza_pod_m261.p3d);
     pylonWeapon = "vtx_dagr";
   };
+  class VTX_M230_Chaingun_L_AP: VTX_M230_Chaingun_L {
```
duplicate

ampersand38 commented on addons/uh60_weapons/config/cfgAmmo.hpp:
```diff
@@ -59,4 +61,34 @@ class CfgAmmo {
+    caliber=4.4;
+    explosive = 0.1;
+    hit = 170;
+    indirectHit = 60;
```
Is this intended, ap having higher `indirectHit` than hedp and same range?

AdozenBABYdogs replied:
There's an issue when it came to hitting components in the vehicles.  The higher indirectHit range made the rounds "punch" more than if they didn't have a higher hit.  This only really comes into play when the round hits a target and is otherwise not capible of injuring ground troops without multipe rounds landing within about 1 meter from a soft target

ampersand38 replied:
I see, and the ace frags makes the HEDP not need a greater range?

AdozenBABYdogs replied:
exactly.  The frags do most of the damage.  Landing a round directly onto someone is the best methood though, but as long as you're in the ballpark of 5-10 meters away, you'll do the job with enough rounds

ampersand38 commented on addons/uh60_weapons/config/cfgMagazines.hpp:
```diff
@@ -45,7 +45,20 @@ class cfgMagazines {
     model = \z\vtx\addons\uh60_weapons\Data\M230\M230_L.p3d;
   };
-  class VTX_M230_Chaingun_R: VTX_M230_Chaingun_L {
+  class VTX_M230_Chaingun_R: VehicleMagazine {
```
why did this need to be explicitly redefined instead of inherited?

AdozenBABYdogs replied:
There was an issue with duplicating the class for AP ammo.  Thinking about a potential solution to the problem now

Commit:
Removed Duplicate

Commit:
Adjusted Inheritance of 30mm ammo

ampersand38 committed:
fix dagr & apkws

ampersand38 committed:
ace_missileguidance_dagr

ampersand38 committed:
remove gatling_20mm

ampersand38 committed:
gun code together, names & desc

ampersand38 committed:
M230 clip on seahawks, 20mm name

ampersand38 committed:
Disable turnout for driver
